project/src/main/nurikabe/fast/fast_solver.gd

```diff
@@ -141,12 +171,32 @@ func enqueue_clued_islands() -> void:
 			_task_queue.append(deduce_clued_island.bind(island_group))
 
 
+func enqueue_island_dividers() -> void:
+	var smallest_island_groups: Array[Array] = board.get_smallest_island_groups()
+	for island_group: Array[Vector2i] in smallest_island_groups:
+		var clue_value: int = board.get_clue_value_for_group(island_group)
+		if clue_value == 0:
+			# unclued island
+			continue
+		var liberties: Array[Vector2i] = board.get_liberties(island_group)
+		if liberties.size() > 0:
+			_task_queue.append(deduce_island_divider.bind(island_group))
+
+
 func enqueue_islands_of_one() -> void:
 	for cell: Vector2i in board.cells:
 		if board.get_cell_string(cell) == "1":
 			_task_queue.append(deduce_island_of_one.bind(cell))
 
 
+func enqueue_wall_expansions() -> void:
+	var smallest_wall_groups: Array[Array] = board.get_smallest_wall_groups()
+	for wall_group: Array[Vector2i] in smallest_wall_groups:
+		var liberties: Array[Vector2i] = board.get_liberties(wall_group)
+		if liberties.size() > 0:
```

Comment:
Stricter check. deduce_wall_expansion currently only handles the single liberty case. If it changes in the future, this check can change too.